examples/1_basic.jl

```diff
@@ -0,0 +1,242 @@
+=#
+
+ForwardDiff.jacobian(_x -> forward_optim(x; method=LBFGS()), x)
+
+# Reverse mode autodiff
+
+Zygote.jacobian(_x -> implicit_optim(_x; method=LBFGS()), x)[1]
+@test Zygote.jacobian(_x -> implicit_optim(_x; method=LBFGS()), x)[1] ≈ J  #src
+
+#=
+In this instance, we cannot use Zygote.jl directly on the solver (due to unsupported `try/catch` statements).
+=#
+
+try
+    Zygote.jacobian(_x -> forward_optim(x; method=LBFGS()), x)[1]
+catch e
+    e
+end
+
+# ## Nonlinear system
+
+#=
+Next, we show how to differentiate through the solution of a nonlinear system of equations:
+```math
+\text{find} \quad y(x) \quad \text{such that} \quad F(x, y(x)) = 0
+```
+The optimality conditions are pretty obvious:
+```math
+F(x, y) = 0
+```
+=#
+
+#=
+To make verification easy, we solve the following system:
+```math
+F(x, y) = y \odot y - x = 0
+```
+In this case, the optimization problem boils down to the componentwise square root function, but we implement it using a black box solver from [NLsolve.jl](https://github.com/JuliaNLSolvers/NLsolve.jl).
+=#
+
+function forward_nlsolve(x; method)
+    F!(storage, y) = (storage .= y .^ 2 - x)
+    initial_y = similar(x)
+    initial_y .= 1
+    result = nlsolve(F!, initial_y; method)
+    return result.zero
+end
+
+#-
+
+function conditions_nlsolve(x, y; method)
+    c = y .^ 2 .- x
+    return c
+end
+
+#-
+
+implicit_nlsolve = ImplicitFunction(forward_nlsolve, conditions_nlsolve)
+
+#-
+
+implicit_nlsolve(x; method=:newton) .^ 2
+@test implicit_nlsolve(x; method=:newton) .^ 2 ≈ x  #src
+
+# Forward mode autodiff
+
+ForwardDiff.jacobian(_x -> implicit_nlsolve(_x; method=:newton), x)
+@test ForwardDiff.jacobian(_x -> implicit_nlsolve(_x; method=:newton), x) ≈ J  #src
+
+#-
+
+ForwardDiff.jacobian(_x -> forward_nlsolve(_x; method=:newton), x)
+
+# Reverse mode autodiff
+
+Zygote.jacobian(_x -> implicit_nlsolve(_x; method=:newton), x)[1]
+@test Zygote.jacobian(_x -> implicit_nlsolve(_x; method=:newton), x)[1] ≈ J  #src
+
+#-
+
+try
+    Zygote.jacobian(_x -> forward_nlsolve(_x; method=:newton), x)[1]
+catch e
+    e
+end
+
+# ## Fixed point
+
+#=
+Finally, we show how to differentiate through the limit of a fixed point iteration:
+```math
+y \longmapsto T(x, y)
+```
+The optimality conditions are pretty obvious:
+```math
+y = T(x, y)
+```
+=#
+
+#=
+To make verification easy, we consider [Heron's method](https://en.wikipedia.org/wiki/Methods_of_computing_square_roots#Heron's_method):
+```math
+T(x, y) = \frac{1}{2} \left(y + \frac{x}{y}\right)
+```
+In this case, the fixed point algorithm boils down to the componentwise square root function, but we implement it manually.
+=#
+
+function forward_fixedpoint(x; iterations)
+    y = ones(eltype(x), size(x))
+    for _ in 1:iterations
+        y .= 0.5 .* (y .+ x ./ y)
+    end
+    return y
+end
+
+#-
+
+function conditions_fixedpoint(x, y; iterations)
```

Comment:
Is there an example where the kwargs are used in the conditions function? if not, then we can consider dropping the kwargs here in another PR.

Reply:
The thing is that the conditions need to accept the same keyword arguments as the forward mapping. We could also do kwargs... here but it's less informative